### ISSUE TITLE
Uniform arguments for similar methods [INCOMPLETE]

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#722]
- [ ] `active_trail_nodes` accepts single nodes as argument whereas `local_independencies` accept both node and iterables.
- [x] `local_independencies` in `BayesianModel` returns an `Independencies` object whereas in `NaiveBayes` it returns a list.
- [ ] `estimator_type` should be replaced by `estimator` in all the `fit`/`predict` methods.
- [ ] Should `get_cpds` accept multiple arguments ?
- [x] `get_cpds` accept an argument but `get_factors` doesn't.
- [ ] `get_cardinality` in `BayesianModel` and `MarkovModel` are different. 
- [ ] Method for finding local independencies is called `get_local_independencies` in `MarkovModel` whereas it's `local_independencies` in `BayesianModel`. Also it accepts arguments in case of `BayesianModel`
- [ ] No `copy` method in `BayesianModel`.
- [ ] `get_cardinality` in `ClusterGraph` and `FactorGraph` doesn't accept any arguments.
- [ ] `get_cardinality` of `Factor` only accepts `lists` and returns `dict` which is different from it's behaviour in other classes..
- [ ] `get_cpd` in `TabularCPD` doesn't seem to be a suitable name for the method.

@khalibartan @yashu-seth If you have noticed any non uniform behavior please add here.

**Comment by chrisittner** [comment]
- [ ] Similar to 4.: I used `prior_type` several times as a keyword argument in `estimators/`, but just `prior` seems nicer. Also `tabu_length`->`tabu`. I'll soon make a PR with small changes for the estimators.

**Comment by raghavg7796** [comment]
I would like to work on this. And in 1.: I think add_edges_from should continue having a list, while add_nodes_from can accept *args

**Comment by ankurankan** [comment]
@raghavg7796 Sure go ahead. But I don't think `add_edges_from` and `add_nodes_from` should have different arguments.

**Comment by raghavg7796** [comment]
@ankurankan Should for 1. 
add_nodes_from and add_edges_from accept *argv 
or add_cpds should accept a iterable?
And changing anyone will require a lot of change in tests...How should I proceed?

**Comment by raghavg7796** [comment]
@ankurankan And in 2. Should active_trail_nodes now return a dict with keys as variables in the iterable given to it and values as sets of active_trail_nodes?

**Comment by raghavg7796** [comment]
@ankurankan Please can you mention which method of the 2 will you prefer in each of the above mentioned issue. I will then create a PR for each of them and PR for 3. has already been created #801

**Comment by yashu-seth** [comment]
@ankurankan I don't think we need to make the ```add_cpds```/```add_factors``` and ```add_nodes_from``` or ```add_edges_from``` accept the same type of input parameter. 
Besides, ```add_nodes_from``` and ```add_edges_from``` behaves the same way in the ```networkx``` classes so don't you think we should leave it as it is?

**Comment by ankurankan** [comment]
@yashu-seth Yes you are right. Plus it won't work in a lot of cases.

**Comment by raghavg7796** [comment]
@ankurankan @yashu-seth In first, I am planning to return a dictionary with keys as nodes. Is it Fine?

**Comment by yashu-seth** [comment]
@raghav7796 In first, (active_trail_nodes accepts ...) you have to change the input parameter not the return type.

**Comment by raghavg7796** [comment]
@yashu-seth Currently active_trail_nodes returns a list of active trail but now we are accepting a list of nodes so we will need to return lists of active trails(active trail for each node). So I was asking about that, that how to return lists of active trail for each nodes

**Comment by yashu-seth** [comment]
@raghavg7796 Sorry. I did not notice that. Yes, I think that would be the best way.

**Comment by lohani2280** [comment]
I think its better to `mark` the sub-issues which have been completed to avoid confusion. For example , issues like  `copy method in BayesianModel` , `get_cpd in TabularCPD doesn't seem to be a suitable name for the method.` have been completed.

**Comment by ankurankan** [comment]
#841 doesn't fix all the points in this issue.
